java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -260,14 +251,16 @@ else if (expression instanceof PsiMethodReferenceExpression) {
 
                 for (int i = 0; i < targetMethodParameters.length; i++) {
                   left[i + offset] = parameters[i + offset].getType();
-                  right[i + offset] = partialSubstitutor.substitute(targetMethodParameters[i].getType());
+                  right[i + offset] = targetMethodParameters[i].getType();
                 }
 
+                final PsiSubstitutor partialSubstitutor = result.getSubstitutor();
```

Comment:
you should not call `result.getSubstitutor()` here, because it relies on the context which is absent here. I've already told you that before.

Reply:
Thank you for your comment, I didn't see your remark about the substitutor, I just went through [the previous PR](https://github.com/JetBrains/intellij-community/pull/1280) and did not find it there. I guess the GitHub interface is overcomplicated. I will bring the generation of the substitutor back.

Reply:
@akozlova , I restored the in-place `PsiSubstitutor` generation for method references with type arguments, please review